torch/nn/attention/flex_attention.py

```diff
@@ -359,8 +363,13 @@ def from_kv_blocks(
             BLOCK_SIZE = (BLOCK_SIZE, BLOCK_SIZE)
 
         mask_mod = mask_mod if mask_mod is not None else noop_mask
+        if seq_lengths is None:
```

Comment:
hmmmmmm idk bout this one

Reply:
I kind of want to preserve the previous flexibility if folks wanted it.

Reply:
![image](https://github.com/user-attachments/assets/5ec2faab-27a8-4ef8-816c-4605410af6ba)

Reply:
@drisspg This doesn't really affect that - it just sets the seq_lengths to the "default" value of num_blocks * block_size.

Reply:
Spoke offline, I disagree but wont die on this hill since this PR gets us meaningfully in a better state